webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,279 @@
+                    <span className='fix-version-label-value'>
+                        {fixVersion}
+                    </span>
+                </div>
+            );
+        }
+        return (<span/>);
+    }
+
+    tagTicketStatus(ticketStatus: string) {
+        if (ticketStatus.toLowerCase() === 'indeterminate') {
+            return (<span className='ticket-status-indeterminate default-style'>{ticketStatus}</span>);
+        }
+
+        if (ticketStatus.toLowerCase() === 'done') {
+            return (<span className='ticket-status-done default-style'>{ticketStatus}</span>);
+        }
+
+        return (<span className='ticket-status-default default-style'>{ticketStatus}</span>);
+    }
+
+    labelList(labels: string[]) {
+        if (labels.length) {
+            let totalString = 0;
+            let totalHide = 0;
+            const labelList = labels.map((label: string, key: number): JSX.Element => {
+                if (totalString < 3) {
+                    totalString++;
+                    return (
+                        <span
+                            key={key}
+                            className='jiraticket-popover-label-list'
+                        >
+                            {label}
+                        </span>);
+                }
+                totalHide++;
+                if (key === labels.length - 1) {
+                    const moreLabels = `+${totalHide}more`;
+                    return (
+                        <span
+                            key={key}
+                            className='jiraticket-popover-label-list'
+                        >
+                            {moreLabels}
+                        </span>);
+                }
+                return <></>;
+            });
+
+            return (<div className={'ticket-popover-label'}>{labelList}</div>);
+        }
+        return null;
+    }
+
+    render() {
+        if (!this.state.isLoaded) {
+            return (<p/>);
+        }
+        const ticketDetails = this.state.ticketDetails;
+        const jiraTicketURI = this.state.href;
+        const jiraAvatar = ticketDetails && ticketDetails.jiraIcon;
+        const jiraIssueIconURI = ticketDetails && ticketDetails.issueIcon;
+        const jiraTicketKey = ticketDetails && ticketDetails.ticketId;
+        const jiraTicketTitle = ticketDetails && ticketDetails.summary;
+        const jiraTicketAssigneeAvatarURI = ticketDetails && ticketDetails.assigneeAvatar;
+        const jiraTicketAssigneeName = ticketDetails && ticketDetails.assigneeName;
+        const jiraTicketStatusName = ticketDetails && ticketDetails.statusKey;
+        const jiraTicketDescription = ticketDetails && ticketDetails.description;
+        const jiraTicketVersions = ticketDetails && ticketDetails.versions;
+        const jiraTicketLabels = ticketDetails && ticketDetails.labels;
+        const isAssigned = ' is assigned';
+        const unAssigned = 'Unassigned';
```

Comment:
These should not be part of the render function. Declare these above the component initialization.